src/vectorclock.erl

```diff
@@ -124,91 +124,33 @@ set_clock_of_dc(DcId, Time, VectorClock) ->
 from_list(List) ->
     dict:from_list(List).
 
-eq(V1, V2) ->
-    dict:fold( fun(Dcid, Time2, Result) ->
-                       case dict:find(Dcid, V1) of
-                           {ok, Time1} ->
-                               case Time1 =:= Time2 of
-                                   true ->
-                                       Result;
-                                   false ->
-                                       false
-                               end;
-                           error ->
-                               false
-                       end
-               end,
-               true, V2).
+-spec for_all_keys(fun((non_neg_integer(), non_neg_integer()) -> boolean()), vectorclock(), vectorclock()) -> boolean().
+for_all_keys(F, V1, V2) ->
+  %% We could but do not care about duplicate DC keys - finding duplicates is not worth the effort
```

Comment:
I might suggest using a sorted list (by DCID).  These would have a smaller memory footprint compared to the dict (less network traffic) and comparisons should then be 2N.  Insertions and single lookups/updates would be slower, but it seems that these are the rare cases.

I think the most common case will be in the materialiser during a read where we have to traverse a  list of operations and snapshots to find the right version to create.  The number of vector clock comparisons you have to make here is only bounded by concurrency and since this is the read path it is probably important to optimise.  The current suggestion looks to have cost 8N for a comparison plus making a copy of the keys of both lists.  While traversing two sorted lists should be 2N.  We would probably have to rework some parts of the materialiser to account for this also though.

I guess it doesn't really matter unless we start getting really big vector clocks. So maybe it is only worth looking at this if we ever find it is an issue.